pkg/hubble/monitor/filter.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	"github.com/cilium/cilium/pkg/hubble/parser/errors"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+)
+
+// monitorFilter is an implementation of OnMonitorEvent interface that filters monitor events.
+type monitorFilter struct {
+	logger logrus.FieldLogger
+
+	drop          bool
+	debug         bool
+	capture       bool
+	trace         bool
+	l7            bool
+	agent         bool
+	policyVerdict bool
+	recCapture    bool
+	traceSock     bool
+}
+
+// NewMonitorFilter creates a new monitor filter.
+// If monitorEventFilters is empty, no events are allowed.
+func NewMonitorFilter(logger logrus.FieldLogger, monitorEventFilters []string) (*monitorFilter, error) {
+	monitorFilter := monitorFilter{logger: logger}
+
+	for _, filter := range monitorEventFilters {
+		switch filter {
+		case monitorAPI.MessageTypeNameDrop:
+			monitorFilter.drop = true
+		case monitorAPI.MessageTypeNameDebug:
+			monitorFilter.debug = true
+		case monitorAPI.MessageTypeNameCapture:
+			monitorFilter.capture = true
+		case monitorAPI.MessageTypeNameTrace:
+			monitorFilter.trace = true
+		case monitorAPI.MessageTypeNameL7:
+			monitorFilter.l7 = true
+		case monitorAPI.MessageTypeNameAgent:
+			monitorFilter.agent = true
+		case monitorAPI.MessageTypeNamePolicyVerdict:
+			monitorFilter.policyVerdict = true
+		case monitorAPI.MessageTypeNameRecCapture:
+			monitorFilter.recCapture = true
+		case monitorAPI.MessageTypeNameTraceSock:
+			monitorFilter.traceSock = true
+		default:
+			return nil, fmt.Errorf("unknown monitor event type: %s", filter)
+		}
+	}
+
+	logger.WithField("filters", monitorEventFilters).Info("Configured Hubble with monitor event filters")
+	return &monitorFilter, nil
+}
+
+func (m *monitorFilter) OnMonitorEvent(ctx context.Context, event *observerTypes.MonitorEvent) (bool, error) {
+	switch payload := event.Payload.(type) {
+	case *observerTypes.PerfEvent:
+		if len(payload.Data) == 0 {
+			return false, errors.ErrEmptyData
```

Comment:
Related to my above comment: `return false` actually passes this event to the hubble subsystem. If you want to not pass empty events (which makes a lot of sense) this should return `true`